src/imp/schannel.rs

```diff
@@ -559,4 +688,119 @@ mod pem {
+            }
+            _ => {}
+        }
+    }
+
+    #[test]
+    fn test_os_provider_os_string() {
+        let my_os_str = OsStr::new("user:RustMy:3e2e13a694b3ed9e40849a4ab98b2c84d1b714d8");
+        let pfx_file = include_bytes!("../test/playserver_openssl2.pfx");
+        let memory_store = PfxImportOptions::new()
+            .include_extended_properties(true)
+            .password("openssl")
+            .import(pfx_file)
+            .unwrap();
+
+        let mut identity = None;
+        for cert in memory_store.certs() {
+            if cert
+                .private_key()
+                .silent(true)
+                .compare_key(true)
+                .acquire()
+                .is_ok()
+            {
+                identity = Some(cert);
+                break;
+            }
+        }
+
+        let identity = identity.unwrap();
+
+        let mut store = CertStore::open_current_user("RustMy").unwrap();
+        store.add_cert(&identity, CertAdd::Always).unwrap();
+
+        let unused_pem:[u8; 0] = [];
+
+        let os_identity = Identity::from_os_provider(&unused_pem, OsStr::new("ncrypt"), my_os_str).unwrap();
+
+        assert_eq!(os_identity.cert.private_key().silent(true).compare_key(true).acquire().is_ok(), true);
+        assert_eq!(identity.fingerprint(HashAlgorithm::sha256()).unwrap(), os_identity.cert.fingerprint(HashAlgorithm::sha256()).unwrap());
+
+        let _result = CertStore::delete_cert_and_key(identity);
+        CertStore::delete_current_user_store("RustMy");
+    }
+
+
+    #[test]
+    fn test_os_provider_sst_file() {
+
+        let pfx_file = include_bytes!("../test/playserver_openssl2.pfx");
+        let mut memory_store = PfxImportOptions::new()
+            .include_extended_properties(true)
+            .password("openssl")
+            .import(pfx_file)
+            .unwrap();
+
+        let dir = env::temp_dir().join("test_sst.sst");
+        let os_str = format!(r"file:{}", dir.to_str().unwrap());     
+        let my_os_str = OsStr::new(os_str.as_str());
+
+
+        CertStore::create_sst(&dir, &mut memory_store).unwrap();
+
+        let mut identity = None;
+        for cert in memory_store.certs() {
+            if cert
+                .private_key()
+                .silent(true)
+                .compare_key(true)
+                .acquire()
+                .is_ok()
+            {
+                identity = Some(cert);
+                break;
+            }
+        }
+        let identity = identity.unwrap();
+
+        let unused_pem:[u8; 0] = [];
+
+        let os_identity = Identity::from_os_provider(&unused_pem, OsStr::new("ncrypt"), my_os_str).unwrap();
+
+        assert_eq!(os_identity.cert.private_key().silent(true).compare_key(true).acquire().is_ok(), true);
+        assert_eq!(identity.fingerprint(HashAlgorithm::sha256()).unwrap(), os_identity.cert.fingerprint(HashAlgorithm::sha256()).unwrap());
+
+        let _result = CertStore::delete_cert_and_key(identity);
```

Comment:
>  let _result = CertStore::delete_cert_and_key(identity); [](http://example.com/codeflow?start=6&length=57)

Save to memory store to sst file with this certificate.

Do another from_os_provider(). It should fail.